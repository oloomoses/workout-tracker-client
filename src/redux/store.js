import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import userRegistrationReducer from './user/userRegisterReducer';

const store = createStore(userRegistrationReducer, applyMiddleware(thunk));

export default store;
