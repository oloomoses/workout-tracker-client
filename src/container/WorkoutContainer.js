import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { workouts } from '../redux/actions';
import Workouts from '../components/Workouts';

const WorkoutContainer = () => {
  const dispatch = useDispatch();
  const date = new Date();

  useEffect(() => {
    dispatch(workouts());
  }, []);

  return (
    <div>
      <div className="text-center">
        Date :
        {date.toString()}
      </div>
      <Workouts />
    </div>
  );
};

export default WorkoutContainer;
