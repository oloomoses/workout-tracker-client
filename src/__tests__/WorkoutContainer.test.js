import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import WorkoutContainer from '../container/WorkoutContainer';
import store from '../redux/store';

describe('WorkoutContainer component', () => {
  it('Matches the snapshot', () => {
    const tree = renderer.create(<Provider store={store}><WorkoutContainer /></Provider>).toJSON;
    expect(tree).toMatchSnapshot();
  });
});
