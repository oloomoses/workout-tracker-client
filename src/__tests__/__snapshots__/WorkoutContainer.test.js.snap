// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutContainer component Matches the snapshot 1`] = `[Function]`;
